plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -240,8 +235,29 @@ const ProductTemplateEdit = (
 			}
 			// If `inherit` is truthy, adjust conditionally the query to create a better preview.
 			if ( inherit ) {
-				if ( templateCategory ) {
-					query.categories = templateCategory[ 0 ]?.id;
+				const { taxonomy, slug } = parseTemplateSlug( templateSlug );
+
+				if ( taxonomy && slug ) {
+					const templateCategory = getEntityRecords(
+						'taxonomy',
+						taxonomy,
+						{
+							context: 'view',
+							per_page: 1,
+							_fields: [ 'id' ],
+							slug,
+						}
+					);
+
+					if ( templateCategory ) {
+						const taxonomyId = templateCategory[ 0 ]?.id;
+						if ( taxonomy === 'category' ) {
+							query.categories = taxonomyId;
+						} else {
+							// If taxonomy is not `category`, we expect either `product_cat` or `product_tag`
```

Comment:
I'm just curious. It will work for other custom taxonomies, too. Right? For example, Brands taxonomy. 🤔 
If yes, maybe we can update the comment to make it more generic.

Reply:
Oh, I see. Thanks for clarifying that. IMO It would be great if we can handle it for other custom taxonomies too. I believe custom taxonomies will have paths like `<<THEME>>//taxonomy-<<TAXONOMY>>-<<TERM>>`; therefore, we should be able to handle them as well. 

But I am not sure if it should be part of this PR. Therefore, We can handle it later in a separate PR. What do you think? 🤔

Reply:
Thanks Karol 🙇 🚀

Reply:
Issue created: https://github.com/woocommerce/woocommerce/issues/50068